### PR TITLE
split selected rows op should always init output selected rows

### PR DESCRIPTION
**Pull request by jacquesqiao**

### CODE REVIEW

File: paddle/fluid/operators/split_selected_rows_op.h [hunk 1]
```diff
@@ -72,10 +72,11 @@ class SplitSelectedRowsOpKernel : public framework::OpKernel<T> {
     for (size_t i = 0; i < outs_rows_idx.size(); ++i) {
       auto rows_idx = outs_rows_idx[i];
       outs[i]->set_height(height_sections[i]);
+      auto dims = x->GetCompleteDims();
+      dims[0] = rows_idx.size();
```
Comment (Yancey0623) [inline paddle/fluid/operators/split_selected_rows_op.h:76]: Does it works well when `rows_ids.size()=0` ?

Comment (jacquesqiao) [reply paddle/fluid/operators/split_selected_rows_op.h:76]: https://github.com/PaddlePaddle/Paddle/blob/00776b167aa5bdfb8a9888f764ec8fa3a0b6f159/paddle/fluid/framework/tensor.cc#L34-L41

Comment (Yancey0623) [reply paddle/fluid/operators/split_selected_rows_op.h:76]: Sure, some element of `outs_rows_idx` may be empty when distributed training, that means that we don't need to call `mutable_data` for that, and the tensor would also not beed initliazed.

**Review by Yancey0623 — DISMISSED** [review]
LGTM

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**